packages/ai/streams/llamaindex-workflow-adapter.ts

```diff
@@ -0,0 +1,106 @@
+import { formatDataStreamPart, JSONValue } from '@ai-sdk/ui-utils';
+import { DataStreamWriter } from '../core/data-stream/data-stream-writer';
+import { createDataStream } from '../core';
+import { prepareResponseHeaders } from '../core/util/prepare-response-headers';
+
+type WorkflowEventData<T> = {
+  data: T;
+};
+
+interface StreamCallbacks {
```

Comment:
@lgrammel we passed DataStreamWriter to let the callback code modify the datastream - might be generally useful for the existing stream callbacks